compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala

```diff
@@ -557,31 +560,31 @@ case class WorkflowCompiler(separateOutputs: Boolean,
         } else {
           irExecutable.inputVars.zip(stage.inputs)
         }
+
         val systemRequirements = irExecutable match {
           // specify the instance type for native app stubs that specify it in the runtime section
-          case Application(
-              _,
-              _,
-              _,
-              StaticInstanceType(
-                  InstanceTypeRequest(Some(staticInstanceType), _, _, _, _, _, _, _, _, _, _)
-              ),
-              _,
-              _: ExecutableKindNative,
-              _,
-              _,
-              _,
-              _,
-              _,
-              _
-              ) =>
-            Some(
-                "systemRequirements" -> JsObject(
-                    "*" -> JsObject(
-                        "instanceType" -> JsString(staticInstanceType)
-                    )
-                )
-            )
+          case app: Application =>
+            val instanceType: String = app.instanceType match {
+              case _: StaticInstanceType
+                  if (app.kind.getClass == classOf[ExecutableKindWfFragment]) =>
+                "do not override"
+              case static: StaticInstanceType => instanceTypeDb.apply(static.req).name
+              case DefaultInstanceType if (app.kind.getClass == classOf[ExecutableKindNative]) =>
+                "do not override"
+              case _ => instanceTypeDb.defaultInstanceType.name
+            }
```

Comment:
Not sure if it should be considered, but in `ApplicationCompiler`
https://github.com/dnanexus/dxCompiler/blob/02a8ed68b27a187c134dded33334ee756c4ceeac/compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala#L178-L187
`defaultInstanceType` is prior to the `instanceTypeDb.defaultInstanceType`. Shall we also assign the stage default here in the same way?

Reply:
In `ApplicationCompiler` the run spec is created, here it's getting overridden by whatever custom instance is requested. Since native app is not compiled by `ApplicationCompiler` - its instance override has to happen later. In this case it was implemented (earlier by someone) for the new instance request to be specified in the stage's `systemRequirements` attr, which presumably has a precedence over `instanceType` attr when the stage is compiled. However, if no overrides were in the WF script, we need to roll back to the native exec's defaults. It's done by NOT specifying `systemRequirements`. I chose to resolve it to None similar to the [`case _` here](https://github.com/dnanexus/dxCompiler/blob/7054ab5b1cdbb90a7162e48dc40b53c18f2054e1/compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala#L588) (for the defaults of the native execs) by this arbitrary string to avoid implementation of something like `dxApi.appdescribe()` -> then parsing a non-trivial nested json that is an app description. But it could be an alternative.

Reply:
I hope that answered your question. If you were asking just to have same cases, that's gonna be difficult because these 2 situations try to resolve different cases, and may have different spectrums of inputs.